src/main.cpp

```diff
@@ -64,70 +45,61 @@ void loop()
   delay(100);
 
   // check if there is a card
-  if (mfrc522.PICC_IsNewCardPresent())
+  if (board.getRfid().PICC_IsNewCardPresent())
   {
     // if there is a "new" card (could be the same that stayed in the field)
-    if (!mfrc522.PICC_ReadCardSerial() || !ready_for_a_new_card)
+    if (!board.getRfid().PICC_ReadCardSerial() || !ready_for_a_new_card)
     {
       return;
     }
     ready_for_a_new_card = false;
 
     // Acquire the UID of the card
-    candidate_user.setUidFromArray(mfrc522.uid.uidByte);
+    auto uid = board.getRfid().uid.uidByte;
 
```

Comment:
In both branches you end up creating FabMember, explicitly in the else, implicitly in the  board.authorize if.

you have a server that authorise, based on the UID, so i guess what you want here is something like

    auto member = server.getMachineMemberFromUID(uid, machineId); //machineId is necessary ONLY if different machine have different user. I assume it return an optional, empty or with valid user
    if (!member){
        //invalid memebr
        return;
    }
    if (board.getMachine().isFree()){
        if (member.isAuthorized()){
            board.setUser(member);
            Serial.println("Login successfull");
        }else{
            Serial.println("Login error");
        }
    }else{
       if (board.getMachine().getActiveUser() == member){ //you created comparison operator, now use it!
          board.resetUser(); //or logout
       }
    }